libs/community/langchain_community/document_loaders/base_o365.py

```diff
@@ -46,24 +48,27 @@ class _O365TokenStorage(BaseSettings):
     token_path: FilePath = Path.home() / ".credentials" / "o365_token.txt"
 
 
-class _FileType(str, Enum):
-    DOC = "doc"
-    DOCX = "docx"
-    PDF = "pdf"
+def fetch_mime_types(file_types: Sequence[str]) -> Dict[str, str]:
+    """Fetch the mime types for the specified file types."""
+    mime_types_mapping = {}
+    for ext in file_types:
+        mime_type, _ = mimetypes.guess_type(f"file.{ext}")
```

Comment:
could we just use a simple dictionary lookup instead?

Reply:
(and same for below)

Reply:
Who's going to maintain that lookup dict? Under the hood that is exactly what `mimetypes` does, except that library _specializes_ in this so their lookup will be always more up to date and complete. Why not offload the conversion to a library?

Reply:
fair point!